packages/lit-html/src/directive-helpers.ts

```diff
@@ -116,14 +116,26 @@ export const insertPart = (
     );
   } else {
     const endNode = wrap(part._$endNode!).nextSibling;
-    const parentChanged = part._$parent !== containerPart;
+    const oldParent = part._$parent;
+    const parentChanged = oldParent !== containerPart;
     if (parentChanged) {
       part._$reparentDisconnectables?.(containerPart);
       // Note that although `_$reparentDisconnectables` updates the part's
       // `_$parent` reference after unlinking from its current parent, that
       // method only exists if Disconnectables are present, so we need to
       // unconditionally set it here
       part._$parent = containerPart;
+      // Since the _$isConnected getter is somewhat costly, only
+      // read it once we know the subtree has directives that need
+      // to be notified
+      let newConnectionState;
+      if (
```

Comment:
Why was this not needed before? Was it just an omission? If so did we add test coverage?

Let's also add a comment here explaining that when a part is moved, its connection state needs to be updated to match the part it's moved into.

Reply:
Not an omission per se; prior to this the user could change any part's connection state, so it was up to the user moving a part to keep it in sync.

With this change, only the root part holds connection state, and everything under it is kept in sync / notified automatically. Added a test in `directive-helpers_test.ts`.